File: Servidor/Calculo-de-Hipotenusa/calc.py
#!C:\Users\zx22student3206\AppData\Local\Programs\Python\Python311\python.exe

print("Content-type: text/html\n")

import os 
from urllib.parse import urlparse, unquote, parse_qs
import math

#Recupera las variables del sistema
ru = os.environ.get("REQUEST_URI")
parsed_url = urlparse(ru)
#Extrae los valores del form de las variables del sistema
parameters = parse_qs(parsed_url[4])

#Convierte el valor de los inputs de string a float
cat1 = float(parameters["cat1"][0])
cat2 = float(parameters["cat2"][0])

#Realiza el cálculo para obtener la hipotenusa
resultado = math.sqrt(cat1**2+cat2**2)

#Imprime el resultado en pantalla
salida = "La hipotenusa mide: {}cm"
print("<html><head/><body><h3>")
print(salida.format("%.2f" % resultado))
print("</h3></body></html>")

File: Cliente/Calculo-de-Hipotenusa/calc.js
function calcula(){
    //Saca los contenidos de los inputs con id "cat1" y "cat2"
    let c1 = document.getElementById("cat1").value;
    let c2 = document.getElementById("cat2").value;
    console.log(c1,c2);

    //Hace el calculo de la hipotenusa con la ayuda de Math.sqrt para hacer la raíz cuadrada
    let resultado = Math.sqrt(c1**2+c2**2);
    console.log(resultado)

    //Imprime el resultado en el html
    document.getElementById("salida").innerHTML = resultado;
}